src/node/blockstorage.h

```diff
@@ -172,6 +172,7 @@ std::ostream& operator<<(std::ostream& os, const BlockfileCursor& cursor);
 
 enum class ReadRawError {
     IO,
+    BadPartRange,
```

Comment:
If you add the value in 0fa70db617a0f003422f66a0ef634b0af1bbd520, you have to handle it in every switch:
https://github.com/bitcoin/bitcoin/actions/runs/20077800373/job/57596718560?pr=33657#step:8:2692

@sedited, are you sure still we need the `assert(false)` hack, it seems `-Wswitch` already warns:
```
/home/runner/work/bitcoin/bitcoin/src/rest.cpp:421:17: error: enumeration value 'BadPartRange' not handled in switch [-Werror,-Wswitch]
  421 |         switch (block_data.error()) {
      |                 ^~~~~~~~~~~~~~~~~~
1 error generated.
```

Reply:
So the -Wswitch wouldn't catch them? I have checked multiple and they all failed the build if I left out any enum value

Reply:
Thanks, I was under the wrong impression this was only the case with bare `enum`s - thanks for the example.